syntax = "proto3";

package yandex.cloud.events.k8s;

import "google/rpc/status.proto";
import "yandex/cloud/events/common.proto";
import "yandex/cloud/validation.proto";

option go_package = "a.yandex-team.ru/cloud/bitbucket/public-api/yandex/cloud/events/k8s;k8s";
option java_package = "yandex.cloud.api.events.k8s";

message CreateNodeGroup {
    Authentication authentication = 1 [(required) = true];
    Authorization authorization = 2 [(required) = true];

    EventMetadata event_metadata = 3 [(required) = true];
    RequestMetadata request_metadata = 4 [(required) = true];

    events.EventStatus event_status = 5 [(required) = true];
    google.rpc.Status error = 6;

    EventDetails details = 7 [(required) = true];

    message EventDetails {
        string cluster_id = 1 [(required) = true];
        string node_group_id = 2 [(required) = true];
    }
}

message UpdateNodeGroup {
    Authentication authentication = 1 [(required) = true];
    Authorization authorization = 2 [(required) = true];

    EventMetadata event_metadata = 3 [(required) = true];
    RequestMetadata request_metadata = 4 [(required) = true];

    events.EventStatus event_status = 5 [(required) = true];
    google.rpc.Status error = 6;

    EventDetails details = 7 [(required) = true];

    message EventDetails {
        string cluster_id = 1 [(required) = true];
        string node_group_id = 2 [(required) = true];
    }
}

message DeleteNodeGroup {
    Authentication authentication = 1 [(required) = true];
    Authorization authorization = 2 [(required) = true];

    EventMetadata event_metadata = 3 [(required) = true];
    RequestMetadata request_metadata = 4 [(required) = true];

    events.EventStatus event_status = 5 [(required) = true];
    google.rpc.Status error = 6;

    EventDetails details = 7 [(required) = true];

    message EventDetails {
        string cluster_id = 1 [(required) = true];
        string node_group_id = 2 [(required) = true];
    }
}

message AutoUpgradeNodeGroup {
    Authentication authentication = 1 [(required) = true];
    Authorization authorization = 2 [(required) = true];

    EventMetadata event_metadata = 3 [(required) = true];
    RequestMetadata request_metadata = 4 [(required) = true];

    events.EventStatus event_status = 5 [(required) = true];
    google.rpc.Status error = 6;

    EventDetails details = 7 [(required) = true];

    message EventDetails {
        string cluster_id = 1 [(required) = true];
        string node_group_id = 2 [(required) = true];
    }
}
